lib/assigner.rb

```diff
@@ -512,28 +510,38 @@ def moderator_post_unassign_action_code(assignment)
     return "unassigned_group#{suffix}" if assignment.assigned_to_group?
   end
 
-  def topic_same_assignee_and_details(assign_to, type, note, status)
-    topic.assignment&.assigned_to_id == assign_to.id &&
-      topic.assignment&.assigned_to_type == type && topic.assignment.active == true &&
-      topic.assignment&.note == note &&
-      (
-        topic.assignment&.status == status ||
-          topic.assignment&.status == Assignment.default_status && status.nil?
-      )
+  def target_same_assignee_and_details(assign_to, type, note, status)
+    return if @target.assignment&.assigned_to_id != assign_to.id
+    return if @target.assignment&.assigned_to_type != type
+    return if !@target.assignment&.active
+    return if @target.assignment&.note != note
+    if @target.assignment&.status != status &&
+         (@target.assignment&.status != Assignment.default_status || status.present?)
+      return
+    end
+
+    true
   end
 
-  def post_same_assignee_and_details(assign_to, type, note, status)
-    @target.is_a?(Topic) &&
-      Assignment
-        .where(topic_id: topic.id, target_type: "Post", active: true)
-        .any? do |assignment|
-          assignment.assigned_to_id == assign_to.id && assignment.assigned_to_type == type &&
-            assignment&.note == note &&
-            (
-              topic.assignment&.status == status ||
-                topic.assignment&.status == Assignment.default_status && status.nil?
-            )
+  def assignee_was_assigned_in_topic(assign_to, type, note, status)
+    Assignment
+      .where(topic_id: topic.id, target_type: "Post", active: true)
+      .any? do |assignment|
+        next if assignment.assigned_to_id != assign_to.id
+        next if assignment.assigned_to_type != type
+        next if assignment.note != note
+        if assignment.status != status &&
+             (assignment.status != Assignment.default_status || status.present?)
+          next
         end
```

Comment:
This should work?

```suggestion
        next if assignment.status != (status.presence || Assignment.default_status)
```

Reply:
It's slightly different, but I just moved everything into a method because I can reuse it.